Subject: Linux-RT 2.6.25.4-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From khilman@mvista.com Wed May 28 15:18:00 2008
Date: Tue, 20 May 2008 15:45:50 -0700
From: Kevin Hilman <khilman@mvista.com>
To: linux-rt-users@vger.kernel.org
Cc: linux-omap@vger.kernel.org, Kevin Hilman <khilman@mvista.com>
Subject: [PATCH 5/5] ARM: OMAP: remove unnecessary locking in clk_get_rate()

The locking in the get_rate() hook is unnecessary, and causes problems
when used with the -rt patch, since it may be called recursively.

Signed-off-by: Kevin Hilman <khilman@mvista.com>
---
 arch/arm/plat-omap/clock.c |    3 ---
 1 file changed, 3 deletions(-)

Index: linux-2.6.25.4-rt4/arch/arm/plat-omap/clock.c
===================================================================
--- linux-2.6.25.4-rt4.orig/arch/arm/plat-omap/clock.c	2008-05-29 09:45:31.000000000 -0400
+++ linux-2.6.25.4-rt4/arch/arm/plat-omap/clock.c	2008-05-29 09:47:33.000000000 -0400
@@ -160,15 +160,12 @@ EXPORT_SYMBOL(clk_get_usecount);
 
 unsigned long clk_get_rate(struct clk *clk)
 {
-	unsigned long flags;
 	unsigned long ret = 0;
 
 	if (clk == NULL || IS_ERR(clk))
 		return 0;
 
-	spin_lock_irqsave(&clockfw_lock, flags);
 	ret = clk->rate;
-	spin_unlock_irqrestore(&clockfw_lock, flags);
 
 	return ret;
 }
